### PR TITLE
Add support for printing "selection" and "current page" and printing code cleanup

### PR DESCRIPTION
**Pull request by vadz**
This incorporates the changes of #23338 and also contains some cleanup to make this code more maintainable in the future.

### CODE REVIEW

**Comment by vadz** [comment]
@sz5000 Is this ok to merge?

File: interface/wx/cmndata.h [hunk 1]
```diff
@@ -535,6 +547,18 @@ class wxPrintDialogData : public wxObject
     */
     bool GetSelection() const;
 
+    /**
+        Returns @true if the user requested that the current page be printed.
+
+        Note that the "current page" is a defined by the application.
```
Comment (PBfordev) [inline interface/wx/cmndata.h:553]:
```suggestion
        Note that the "current page" is defined by the application.
```
Or, a noun should be added after the "a".

**Review by PBfordev — COMMENTED** [review]
Just two typos in comments.

**Review by sz5000 — APPROVED** [review]
I think that's OK.

**Comment by vadz** [comment]
> Just two typos in comments.

Thanks, force pushed now to fix them.

**Pull request merged**